cmake/modules/FindSpdlog.cmake

```diff
@@ -0,0 +1,109 @@
+# FindSpdlog
+# -------
+# Finds the Spdlog library
+#
+# This will define the following variables:
+#
+# SPDLOG_FOUND - system has Spdlog
+# SPDLOG_INCLUDE_DIRS - the Spdlog include directory
+# SPDLOG_LIBRARIES - the Spdlog libraries
+# SPDLOG_DEFINITIONS - the Spdlog compile definitions
+#
+# and the following imported targets:
+#
+#   Spdlog::Spdlog   - The Spdlog library
+
+if(PKG_CONFIG_FOUND)
+  pkg_check_modules(PC_SPDLOG spdlog QUIET)
+  set(SPDLOG_VERSION ${PC_SPDLOG_VERSION})
+endif()
+
+find_path(SPDLOG_INCLUDE_DIR NAMES spdlog/spdlog.h
+                             PATHS ${PC_SPDLOG_INCLUDEDIR})
+
+if (NOT ENABLE_INTERNAL_SPDLOG AND NOT SPDLOG_INCLUDE_DIR)
+  set(ENABLE_INTERNAL_SPDLOG ON)
```

Comment:
Why auto fall-back to the internal build?
It should error if it is not found so user can then decide to install the (distribution) library or explicitly enable the internal build.

Reply:
Please read the discussion in this PR. I asked for guidance on how to handle this and @lrusak said I should do it this way. Furthermore for spdlog there is currently no way to install the library through Ubuntu packages because the provided packages are too old and incompatible.

Reply:
I'm not telling you to remove the internal build, only to remove the auto-fallback.
So if you haven't installed spodlog, your distribution doesn't provide spdlog or ships and older version them configure. Then you can decide to install it or configrure with internal spdlog, like it's currently possible for flatbuffers, fmt, fstrcmp and rapidjson.

Reply:
For Ubuntu the situation with `libspdlog-dev` is as follows:
* 16.04 provides a version which doesn't actually exist. Nobody knows where it comes from. This version is incompatible.
* 18.04 provides an older incompatible version.
* 19.04 provides an older incompatible version.
* 20.04 will provide a compatible version.

So the situation is similar to `libfmt-dev`. But please discuss this with @lrusak since he suggested to do this auto-fallback. I don't have any opinion on depends builds since I usually work on Windows.

Reply:
I agree with Rechi, auto-fallback is not needed, specially since it doesn't work for packaging. Users should be aware what happens and decide for themselves how they want to proceed.

For ubuntu: I'll have to backport libspdlog first

Reply:
I removed the fallback to internal build.